### PR TITLE
Add tests to improve coverage

### PR DESCRIPTION
**Pull request by blag**
Add additional tests to improve coverage of `scripts/stackstorm.js`.

Note that the second commit removes a test. It seems to cause test failures for other tests, so I left in the git history for somebody else to possibly get working but it didn't bump the coverage much more.

Before these tests, coverage was:

```
------------------------|----------|----------|----------|----------|-------------------|
File                    |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------|----------|----------|----------|----------|-------------------|
All files               |    85.78 |    81.29 |    85.59 |    85.37 |                   |
 lib                    |       99 |    99.57 |    98.86 |    98.96 |                   |
  command_factory.js    |      100 |      100 |      100 |      100 |                   |
  format_command.js     |      100 |      100 |      100 |      100 |                   |
  format_data.js        |      100 |      100 |      100 |      100 |                   |
  post_data.js          |      100 |      100 |      100 |      100 |                   |
  slack-messages.js     |      100 |      100 |      100 |      100 |                   |
  slack_monkey_patch.js |    28.57 |       80 |       50 |    28.57 |    17,18,19,40,41 |
  utils.js              |      100 |      100 |      100 |      100 |                   |
 scripts                |    50.79 |     42.2 |    46.67 |    50.79 |                   |
  stackstorm.js         |    50.79 |     42.2 |    46.67 |    50.79 |... 11,443,444,451 |
------------------------|----------|----------|----------|----------|-------------------|
```

With these tests, coverage is now:

```
------------------------|----------|----------|----------|----------|-------------------|
File                    |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------|----------|----------|----------|----------|-------------------|
All files               |    89.11 |    87.13 |    89.83 |    88.81 |                   |
 lib                    |       99 |    99.57 |    98.86 |    98.96 |                   |
  command_factory.js    |      100 |      100 |      100 |      100 |                   |
  format_command.js     |      100 |      100 |      100 |      100 |                   |
  format_data.js        |      100 |      100 |      100 |      100 |                   |
  post_data.js          |      100 |      100 |      100 |      100 |                   |
  slack-messages.js     |      100 |      100 |      100 |      100 |                   |
  slack_monkey_patch.js |    28.57 |       80 |       50 |    28.57 |    17,18,19,40,41 |
  utils.js              |      100 |      100 |      100 |      100 |                   |
 scripts                |    62.96 |    60.55 |    63.33 |    62.96 |                   |
  stackstorm.js         |    62.96 |    60.55 |    63.33 |    62.96 |... 09,410,411,444 |
------------------------|----------|----------|----------|----------|-------------------|
```

Please note that `scripts/stackstorm.js` is also covered by the ChatOps end-to-end tests in the st2tests repository.

### CODE REVIEW

File: tests/test-st2bot-envvars.js [hunk 1]
```diff
@@ -82,15 +88,16 @@ describe("environment variable configuration", function () {
     // Load script under test
     var stackstorm = require("../scripts/stackstorm.js");
     stackstorm(robot).then(function (stop) {
-      expect(info_spy).to.have.been.calledThrice;
+      expect(info_spy.args).length.to.be.above(1);
+      // expect(info_spy).to.have.been.calledWith('Using ST2_API_KEY as authentication. Expiry will lead to bot exit.');
```
Comment (armab) [inline tests/test-st2bot-envvars.js]: Do we need this line commented or it was just something temporary?
Comment like that makes it look like unfinished.

File: tests/test-st2bot-envvars.js [hunk 2]
```diff
@@ -82,15 +88,16 @@ describe("environment variable configuration", function () {
...
       expect(info_spy).to.have.been.calledWith('Two-factor auth is enabled');
       expect(info_spy).to.have.been.calledWith('Loading commands....');
 
       stop();
 
       done();
     }).catch(function (err) {
-      console.log(err);
+      console.log(err.message);
```
Comment (armab) [inline tests/test-st2bot-envvars.js]: Any specific reason using `err.message`? Could that be `err` as it was before and for consistency with other tests?

File: tests/test-st2bot-sigusr2.js [hunk 3]
```diff
@@ -51,16 +78,16 @@ describe("SIGUSR2", function () {
...
 
       done();
-    }).catch(function (err) {
-      console.log(err);
-      stop && stop();
-      done(err);
+    // }).catch(function (err) {
+    //   console.log('err: ' + err);
+    //   stop && stop();
+    //   done(err);
```
Comment (armab) [inline tests/test-st2bot-sigusr2.js]: Is this code WIP?

File: tests/test-st2bot-sigusr2.js [hunk 4]
```diff
@@ -35,9 +36,35 @@ describe("SIGUSR2", function () {
   var debug_spy = sinon.spy(robot.logger, 'debug'),
     info_spy = sinon.spy(robot.logger, 'info');
 
+  beforeEach(function () {
+    // emulate ST2 API response
+    nock('http://localhost:9101')
```
Comment (armab) [inline tests/test-st2bot-sigusr2.js]: Seems the test worked before without Nock and emulating HTTP response.
What's the advantage of adding it now? Could be confusing a bit, giving assumption that specific HTTP response is required for `SIGUSR2` signal to work properly.

Comment (blag) [reply tests/test-st2bot-sigusr2.js]: It increases code coverage by taking paths in `loadCommands` that other tests do not.

I'll add a comment that it isn't required.

File: tests/test-st2bot-envvars.js (hunk 1 above)
Comment (blag) [reply tests/test-st2bot-envvars.js]: Fixed, thanks.

File: tests/test-st2bot-envvars.js (hunk 2 above)
Comment (blag) [reply tests/test-st2bot-envvars.js]: Nope, fixed to be more consistent.

File: tests/test-st2bot-sigusr2.js (hunk 3 above)
Comment (blag) [reply tests/test-st2bot-sigusr2.js]: Nope, removed.

**Comment by blag** [comment]
It's already in the [changelog (line 8)](https://github.com/StackStorm/hubot-stackstorm/edit/add-tests/CHANGELOG.rst):

```
* Small refactor and more tests (for `scripts/stackstorm.js`) (improvement)
```

**Pull request merged**